src/loaders/compute_tools/mash/mash.py

```diff
@@ -17,21 +19,37 @@ def _run_mash_single(
         debug: bool,
         kmer_size: int = KMER_SIZE,
         sketch_size: int = SKETCH_SIZE) -> None:
+
     # RUN mash sketch for a single genome
-    command = ['mash', 'sketch',
-               '-o', source_file,  # Output prefix.
-               # Save result file to source file directory. The suffix '.msh' will be appended.
-               '-k', f'{kmer_size}',
-               '-s', f'{sketch_size}',
-               source_file]
-
-    run_command(command, output_dir / f'{genome_id}' if debug else None)
-
-    # Save run info to a metadata file in the output directory for parsing later
-    metadata_file = output_dir / f'{genome_id}' / f'mash_run_metadata.json'
-    metadata = {'source_file': str(source_file), 'kmer_size': kmer_size, 'sketch_size': sketch_size}
-    with open(metadata_file, 'w') as f:
-        json.dump(metadata, f, indent=4)
+
+    remove_gz_file = False
+    if source_file.suffix == '.gz':
+        remove_gz_file = True
+        source_file = unpack_gz_file(source_file)
```

Comment:
I think we should build this stuff into the `ToolRunner.parallel_*` methods as it seems like a pretty common use case. Could just add an `unzip=False` default param to the methods

Reply:
Maybe do that in a different PR? If you want to do it here that's also fine

Reply:
updated runner methods.

Reply:
IIUC, the meta file will now contain the unzipped file as the source, but that unzipped file gets deleted? That seems wrong if so

Reply:
right. But that is the true source file used to execute the tool. And if we are going to use source file as some identifier later, we need to know the file that the tool consumes. i.e. checkm2 uses source file name without extension as genome identifier.

Reply:
Ugh. I can see this causing issues in the future. Maybe we need `source` and `uncompressed_source` with the understanding that if `uncompressed_source` exists that was provided to the tool rather than `source`?

Reply:
👍